crates/ignore-files/src/filter.rs

```diff
@@ -75,108 +90,173 @@ impl IgnoreFilter {
 		// TODO: different parser/adapter for non-git-syntax ignore files?
 
 		trace!(files=%files_contents.len(), "building ignore list");
-		let mut builder = GitignoreBuilder::new(origin);
+
+		let mut ignores_trie = Trie::new();
+
+		// add builder for the root of the file system, so that we can handle global ignores and globs
+		ignores_trie.insert(
+			prefix(origin),
+			Ignore {
+				gitignore: Gitignore::empty(),
+				builder: Some(GitignoreBuilder::new(origin)),
+			},
+		);
+
+		let mut total_num_ignores = 0;
+		let mut total_num_whitelists = 0;
+
 		for (file, content) in files_contents.into_iter().flatten() {
 			let _span = trace_span!("loading ignore file", ?file).entered();
+
+			let applies_in = get_applies_in_path(origin, Some(&file));
+
+			let parent_ignore = ignores_trie
+				.get_ancestor_value(&applies_in.display().to_string())
+				// unwrap will always succeed because we created an entry with the root of the origin
+				.unwrap();
```

Comment:
I'm encountering this `unwrap` after updating to `ignore-files` `1.3.0` in previously-working code. This is what I was using before:

```rust
    let (ignores, errors) = ignore_files::from_origin(".").await;
    let mut filt = ignore_files::IgnoreFilter::new([origin](std::env::current_dir()?), &ignores).await?;
```

And it looks like the issue is fixed by using an absolute path in `from_origin`:

```rust
    let origin = std::env::current_dir()?;
    let (ignores, errors) = ignore_files::from_origin(&origin).await;
    let mut filt = ignore_files::IgnoreFilter::new(origin, &ignores).await?;
```

It seems that there was a breaking change to the API where you could previously pass `[IgnoreFile]` with relative paths to `IgnoreFilter::new`, but now this does not work with the new radix trie implementation. 

Anyway, I've fixed my issue, but just wanted to let you know and document for others who may run into the same thing.

Reply:
Ah, thanks for the report! I'll get that fixed up, release a .1, and yank the current version (in this order).

Reply:
oh shoot, sorry about that. @passcod let me know if you want me to fix it, if you're too busy.

Reply:
I'd appreciate it @Cammisuli ! I'm really flat out elsewhere currently 😓

Reply:
1.3.0 yanked as promised now that 1.3.1 has been released.